packages/types/src/action.ts

```diff
@@ -75,12 +77,9 @@ export type AsyncAction<Pkg extends Package, Input = null> = [Input] extends [
       libraries: Pkg["libraries"];
     }) => (args: Input) => Promise<void>;
 
-type Context = {
-  ctx: Koa.ParameterizedContext<Koa.DefaultState, Koa.DefaultContext>;
-};
+export type Context = Koa.ParameterizedContext<
+  Koa.DefaultState,
+  Koa.DefaultContext
+>;
 
-export type ServerAction<Pkg> =
-  | Action<Pkg>
-  | Action<Pkg, Context>
-  | AsyncAction<Pkg>
-  | AsyncAction<Pkg, Context>;
+export type ServerAction<Pkg> = AsyncAction<Pkg, { ctx: Context }>;
```

Comment:
@luisherranz You wrote in the commit message: 
> I think that we can leave ServerAction as both async and requiring a Context for the moment.

I thought that it all 4 should be possible:
- sync, no `ctx`
- sync, with `ctx`
- async, no `ctx`
- async, with `ctx`
But with the type that you added, TS will only allow "async with ctx". Is there something I'm missing?

See the comment above for more details.

Reply:
Yes, let's leave it like that for the moment. If people want to use other combinations in their packages they can use `Action` and `AsyncAction` when defining the types of their `beforeSSR` action.